hist/hist/src/TAxis.cxx

```diff
@@ -404,6 +397,29 @@ Int_t TAxis::FindFixBin(const char *label) const
    return -1;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Internal function to find the fix bin
+////////////////////////////////////////////////////////////////////////////////
+Int_t TAxis::DoFindFixBin(Double_t x) const
+{
+   int bin = 0;
+   if (!fXbins.fN) {        //*-* fix bins
+      // shift x and fXmax by fXmin:
+      x = x - fXmin;
+      const double b = fXmax - fXmin;
+      const double s = fNbins * x; // scaled version of x
+      double m = std::floor(s / b);
+      // iterative correction in case of floating point errors due to floating point division
+      while (m * b >= s)
+         m = m - 1;
+      while ((m + 1) * b < s)
+         m = m + 1;
```

Comment:
```suggestion
      while (m * b > s) // if left bin boundary is greater then x, decrement
         m = m - 1;
      while ((m + 1) * b <= s) //if right bin boundary is smaller or equal, increment
         m = m + 1;
```
Sorry, I did the typical edge case mistake! I understand now that if the value is on the lower boundary, it should be considered in the bin.